cypress/integration/menu/position.feature

```diff
@@ -1,42 +1,46 @@
 Feature: Position of a menu component
 
+    # default max width: 380px; default max height: 380px
     Background:
-        # the default maximum
-        Given the menu component has a height of 368px
-        # also the default maximum
-        And the menu component has a width of 380px
-        And the menu component is triggered by a user click on an anchor element
+        Given the menu component has a height and height of its default maximum
 
     Scenario: Default rendering
-        Given the anchor has more than 368px space below, before the end of the <body> element
-        When the menu is rendered
-        Then the menu is rendered below the anchor
+        Given there is enough space below the anchor to fit the default maximum
```

Comment:
I do prefer your version, it is much simpler. However, is it too generic? 'Enough space below the anchor' seems ambiguous to me, enough space before what? The viewport, the body, another element?

Reply:
It would be nice to base ourself on the dimensions of the actual content, instead of the maximum. Which is difficult since we don't know those before rendering.

There are ways to test the dimensions, by rendering invisibly and then measuring, but that would be slower (and maybe slightly convoluted). I do think it's good to discuss this though.

Just basing ourselves on the maximum means a component with barely any content would still base itself on the maximum and might render in a different direction unnecessarily. You can't correct it afterwards, because then it'd move around, confusing the user.

Reply:
@cooper-joe Feature files shouldn't really contain design specifics, they should define behavior, which should be always the same, regardless of the design. It's not always possible to separate these two, but using pixel values is an anti pattern in my eyes. These values will be defined on the component & could change easily. The tests themselves will use these values (i. e. by reading them from the component's default props). So we're testing with real values, we're just keeping them out of the functional specs.

> The viewport, the body, another element?

This is a bit different. We can include it, I'm indecisive with this one. Let's discuss that in the meeting today?

Reply:
> EDIT: Ah, so I assume that here we're stating that the contents are already at least at the maximum (the menu component has a height and height of its default maximum)? Still, how would we know that before rendering?

@ismay Yes. So far the popover component I've used just uses the dimensions after initial rendering and then moved the component, no glitches so far. But if we find out there are glitches, we can just use set `visibility: hidden` and add a class name `positioned` to remove that style after its position has been determined

Reply:
@cooper-joe @ismay ping

Reply:
I'm happy with your suggestions here. The decision needed, I think is: 
> The viewport, the body, another element?
I think `<body>` is the best to use, as I am happy for a menu/other component to be positioned out of view where it can be scrolled into view, but not out of the `<body>` where it is inaccessible.

Reply:
So we've agreed that we want to use the body as the frame we want to position the components inside. Is it still ok to you (@cooper-joe) to keep the steps as vague as they are in this PR?

Reply:
ping

Reply:
I'm happy to defer that decision to someone with more experience writing effective tests but to me it is required that we state 'the body' here, or somewhere, to avoid ambiguity. If we don't state it in the test steps, how will we always remember it's to 'body' and not 'viewport'?

Reply:
Good point @cooper-joe. After having thought about this for a bit, I think we should include "body" or "viewport" as that's part of the behavior, and then it'll be documented as well. I'll add that to the spec files now

Reply:
Done, see https://github.com/dhis2/ui-core/pull/527#issuecomment-562426939